custom_components/google_home/models.py

```diff
@@ -153,6 +182,327 @@ def as_dict(self) -> GoogleHomeTimerDict:
         }
 
 
+class GoogleHomeBTDevice:
+    """Local representation of detected BT devices"""
+
+    def __init__(
+        self,
+        mac_address: str,
+        device_class: int,
+        device_type: int,
+        rssi: int,
+        expected_profiles: int,
+        name: str | None,
+    ) -> None:
+
+        self.mac_address = mac_address
+        self.device_class = device_class
+        self.device_type = device_type
+        self.rssi = rssi
+        self.expected_profiles = expected_profiles
+        self.name = name
+
+    def as_dict(self) -> GoogleHomeBTDeviceDict:
+        """Return typed dict representation."""
+        return {
+            "mac_address": self.mac_address,
+            "device_class": self._decode_device_class(),
+            "device_type": self._decode_device_type(self.device_type),
+            "rssi": self.rssi,
+            "expected_profiles": self.expected_profiles,
+            "name": self.name,
+        }
+
+    @staticmethod
+    def _decode_device_type(device_type: int) -> str:
+        types = ["BREDR", "BLE"]
+
+        out_types = []
+
+        for num, name in enumerate(types):
+            if device_type & (1 << num):
+                out_types.append(name)
+
+        return "|".join(out_types)
+
+    @staticmethod
+    def _device_major_class(major_number: int) -> str:
```

Comment:
You include a lot of BT specific code. Aren't there any libs that can help with this? The reason I ask is that this will increase the maintain burden if something changes in the future.